src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ExportInterceptingPropertyValueProvider]
+    internal sealed class TargetFrameworkValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+
+        [ImportingConstructor]
+        public TargetFrameworkValueProvider(IUnconfiguredProjectVsServices projectVsServices)
+        {
+            Requires.NotNull(projectVsServices, nameof(projectVsServices));
+
+            _projectVsServices = projectVsServices;
+        }
+
+        public override string GetPropertyName() => "TargetFramework";
+
+        public override async Task<string> OnGetEvaluatedPropertyValueAsync(string evaluatedPropertyValue, IProjectProperties defaultProperties)
+        {
+            Requires.NotNull(_projectVsServices.VsHierarchy, "vsHierarchy");
+
+            // Fetch the target framework version from the VSHierarchy.
+            object targetFrameworkVersion;
+            if (ErrorHandler.Succeeded(_projectVsServices.VsHierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)Shell.VsHierarchyPropID.TargetFrameworkVersion, out targetFrameworkVersion)))
+            {
+                return ((uint)targetFrameworkVersion).ToString();
```

Comment:
Yes, but even that is not an integral value. Its of the format `<TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>`. "TargetFramework" seems to be a derived property created by taking the moniker and then performing [bit manipulation](http://index/#Microsoft.VisualStudio.ProjectSystem.VS.Implementation/Package/Nodes/ProjectNode.Hierarchy.cs,765) with the framework version - CPS already has an API on IUnconfiguredVsProjectServices.VsHierarchy to fetch this evaluated value in a non-UI thread way, which I have used here.